serwer.c

```diff
@@ -0,0 +1,187 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/ipc.h>
+#include <unistd.h>
+#include <sys/shm.h>
+#include <sys/sem.h>
+#include <signal.h>
+
+#define MY_MSG_SIZE 100 //Maksymalny rozmiar wiadomosci
+#define LOGIN_SIZE 20 //Maksymalny rozmiar loginu uzytkownika
+
+key_t shm_entries_key;
+key_t shm_state_key;
+key_t sem_entries_key;
+key_t sem_counter_key;
+int shm_entries_id;
+int shm_state_id;
+int sem_entries_id;
+int sem_counter_id;
+
+//Struktura dla wpisu
+struct entry{
+	char username[LOGIN_SIZE];
+	char message[MY_MSG_SIZE];
+	int likes;
+};
+
+//Struktura dla stanu serwera
+struct state{
+	int n; //Maksymalna liczba wpisow
+	int counter; //Aktualna liczba wpisow
+};
```

Comment:
Rozważ przeniesienie deklaracji nowych typów powyżej deklaracji pól

```suggestion
//Struktura dla wpisu
struct entry{
	char username[LOGIN_SIZE];
	char message[MY_MSG_SIZE];
	int likes;
};

//Struktura dla stanu serwera
struct state{
	int n; //Maksymalna liczba wpisow
	int counter; //Aktualna liczba wpisow
};

key_t shm_entries_key;
key_t shm_state_key;
key_t sem_entries_key;
key_t sem_counter_key;
int shm_entries_id;
int shm_state_id;
int sem_entries_id;
int sem_counter_id;
```